### PR TITLE
Feature: Card Start Date

### LINKED ISSUE #68: Card start date
Along with the due date, we need to be able to set a card's start date. This will be crucial for gantt and calendar views

### PR DESCRIPTION
**Pull request by codonade**
Set a starting date for a card along side its due date. Fixes #68

### CODE REVIEW

**Comment by codonade** [comment]
Now, there are two main issues I know of: You can set `startsAt` to be after `dueAt` or vice versa, set `dueAt` to be before `startsAt`, and changes to fields don't sync between views until you refresh the application. Regarding the first issue, I didn't know where should I handle the validation; on the client or on the server?

File: packages/frontend/src/components/project-management/views/BaseView.vue [hunk 1]
```diff
@@ -65,6 +64,12 @@ const columns = ref<ColumnDef<ListGroup, any>[]>([
     header: 'Assignee',
     size: 100,
   },
+  {
```
Comment (MoShizzle) [inline packages/frontend/src/components/project-management/views/BaseView.vue:67]: Remove it from the views main page, only make it show on the BaseCard

**Review by MoShizzle — CHANGES_REQUESTED** [review]

**Comment by MoShizzle** [comment]
> Now, there are two main issues I know of: You can set `startsAt` to be after `dueAt` or vice versa, set `dueAt` to be before `startsAt`, and changes to fields don't sync between views until you refresh the application. Regarding the first issue, I didn't know where should I handle the validation; on the client or on the server?

That's fine we don't have to handle it right now.

Regarding validation, usually you'd do both frontend and backend validation

File: packages/frontend/src/components/project-management/views/BaseView.vue (hunk 1 above)
Comment (codonade) [reply packages/frontend/src/components/project-management/views/BaseView.vue:67]: So, if I got you correctly, you want the start date property to only be visible when you specifically open the card, and not show it in other views?

Comment (MoShizzle) [reply packages/frontend/src/components/project-management/views/BaseView.vue:67]: Yes exactly

**Comment by codonade** [comment]
Here it is, I reverted the repository status to when you could only edit the start date from the card view. I hope I haven't missed anything up with what I've did...

File: packages/frontend/src/components/project-management/cards/BaseCard.vue [hunk 2]
```diff
@@ -253,6 +253,13 @@ function updateCardListStage(listStage: ListStage) {
           show-first-names
           label="Assign"
         />
+        <base-date-picker
+          class="text-body-2 ms-n2"
+          label="Due date"
```
Comment (MoShizzle) [inline packages/frontend/src/components/project-management/cards/BaseCard.vue:258]: This should be Start date

**Review by MoShizzle — CHANGES_REQUESTED** [review]

**Review by MoShizzle — APPROVED** [review]

**Pull request merged**